File: raster/src/main/java/com/kjipo/raster/segment/Pair.java
package com.kjipo.raster.segment;

public class Pair {
    private final int row;
    private final int column;

    public Pair(int row, int column) {
        this.row = row;
        this.column = column;
    }

    public int getRow() {
        return row;
    }

    public int getColumn() {
        return column;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Pair pair = (Pair) o;

        if (row != pair.row) return false;
        return column == pair.column;
    }

    @Override
    public int hashCode() {
        int result = row;
        result = 31 * result + column;
        return result;
    }
}

File: common/src/main/kotlin/representation/segment/Pair.kt
package com.kjipo.representation.segment

data class Pair(val row: Int, val column: Int) {

    override fun toString(): String {
        return "Pair{" +
                "row=" + row +
                ", column=" + column +
                '}'
    }

    companion object {
        fun of(row: Int, column: Int): Pair {
            return Pair(row, column)
        }
    }
}